api/src/YoutubeClient.cpp

```diff
@@ -29,6 +29,10 @@ std::string VideosResult::Video::GetMediumThumbnail() const {
   return json::StringRef(data_["snippet"]["thumbnails"]["medium"]["url"]);
 }
 
+bool VideosResult::Video::IsAgeRestricted() const { 
+  return data_["contentDetails"]["contentRating"]["ytRating"] == "ytAgeRestricted";
```

Comment:
comparing strings like this using rapidjson doesn't work iirc... you might have to use the `json::StringRef` utility like this https://github.com/MemeLabs/Rustla2/blob/master/api/src/SmashcastClient.cpp#L48